Unique items support

AlexanderKomi opened the pull request:
an attempt to to fix issue: https://github.com/cjbooms/fabrikt/issues/387

cjbooms commented on src/test/kotlin/com/cjbooms/fabrikt/generators/ModelGeneratorTest.kt:
```diff
@@ -68,6 +68,7 @@ class ModelGeneratorTest {
         "byteArrayStream",
         "untypedObject",
         "primitiveTypes",
+        "uniqueItems"
```
Rather than add a completely new test case and spec, just add an array with this property set to true on an existing test spec.

AlexanderKomi replied:
I moved the test case to the array generation test cases.
Also I added a test case to check if an property with the "uniqueItems" flag does generate a Set.
Unfortunately this does not work and I have no idea why.

cjbooms commented:
Nice work, great little addition

AlexanderKomi commented:
Unfortunately my changes do not have the intended effect. 
Currently it does not work and I have no idea why.
The tests fail at this point and I will leave it as it is.
Someone should feel free to change whatever is necessary to make it work.

AlexanderKomi commented:
I rebased on the master branch and fixed the implementation to generate sets.
@cjbooms Please take a look at the implementation and suggest changes, if any are required.
Not sure if the README.md should include this features aswell.

cjbooms commented:
Great, I should be able to get a release out this evening for you. nice work
